server/src/main/java/org/apache/uniffle/server/ShuffleFlushManager.java

```diff
@@ -142,87 +143,96 @@ public void addToFlushQueue(ShuffleDataFlushEvent event) {
     }
   }
 
-  private void flushToFile(ShuffleDataFlushEvent event) {
-
-    Storage storage = storageManager.selectStorage(event);
-    if (storage != null && !storage.canWrite()) {
-      addPendingEvents(event);
-      return;
-    }
-
+  private void flushToFileImpl(ShuffleDataFlushEvent event) {
     long start = System.currentTimeMillis();
-    List<ShufflePartitionedBlock> blocks = event.getShuffleBlocks();
     boolean writeSuccess = false;
-    try {
```

Comment:
Why do we need to modify the flush process?

Reply:
I think this logic need to be refactored.